### PR TITLE
Implement the Rabin fingerprint algorithm

### PR DESCRIPTION
**Pull request by jeffw-wherethebitsroam**
This adds Rabin fingerprinting via Digest.

I haven't done too much rust, so let me know if anything needs changing.

### CODE REVIEW

**Comment by poros** [comment]
Uh, interesting. I honestly don't know how Rabin fingerprint works, but I'll have a look anyway 😉

File: src/rabin.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    };
+}
+
+/// Implementation of the Rabin fingerprint algorithm
+#[derive(Clone)]
+pub struct Rabin {
+    result: i64,
+}
+
+impl Default for Rabin {
```
Comment (poros) [inline src/rabin.rs:27]: What's the reason for not implementing the `Digest` trait? On the [official documentation](https://docs.rs/digest/0.9.0/digest/trait.Digest.html)
> It's a convenience wrapper around Update, FixedOutput, Reset, Clone, and Default traits. It also provides additional convenience methods.

Also, from our definition of fingerprint: https://github.com/flavray/avro-rs/blob/b3edfacced605a9d3229426441a2808176c86120/src/schema.rs#L416

**Review by poros — CHANGES_REQUESTED** [review]
Thanks for contributing! I can't vouch for the correctness of the algorithm, but the tests should work as sanity check.

Could you also update the docs please? https://github.com/flavray/avro-rs/blob/b3edfacced605a9d3229426441a2808176c86120/src/schema.rs#L416

You can then sync them to the README by running `make readme`.

It would also be great if you could add a line in the CHANGELOG.

File: src/rabin.rs (hunk 1 above)
Comment (jeffw-wherethebitsroam) [reply src/rabin.rs:27]: digest has an implementation of `Digest` if you implement the traits I implemented:
```
impl<D: Update + FixedOutput + Reset + Clone + Default> Digest for D
```
And that implements a few of the functions for you, e.g. `chain`, `output_size`, `digest`. So it seems easier. 
Also, I was looking at the `Digest` implementation in the `sha2` crate, and that is how they do it.

**Comment by jeffw-wherethebitsroam** [comment]
I think I've done all of the docs and CHANGELOG updates now.

The algorithm is converted from the java version at https://github.com/apache/avro/blob/master/lang/java/avro/src/main/java/org/apache/avro/SchemaNormalization.java

So it should be correct. And like you say, should be verified by the shared tests from https://github.com/apache/avro/blob/master/share/test/data/schema-tests.txt

File: src/rabin.rs (hunk 1 above)
Comment (poros) [reply src/rabin.rs:58]: I see. Thanks for the explanation.

**Review by poros — APPROVED** [review]
Looks good to me. Thanks again for contributing!

**Pull request merged**